src/test/cbmc/SIMD/Construction/main.rs

```diff
@@ -36,4 +43,17 @@ fn main() {
         // Original unchanged
         assert!(y.0 == 0 && y.1 == 1);
     }
+    {
+        const I: [u32; 2] = [1, 2];
+        let x: i64x2 = unsafe { simd_shuffle2(y, z, I) };
+        assert!(x.0 == 1);
+        assert!(x.1 == 1);
+    }
+    {
+        let a = i64x4(1, 2, 3, 4);
+        let b = i64x4(5, 6, 7, 8);
+        const I: [u32; 4] = [1, 3, 5, 7];
+        let c: i64x4 = unsafe { simd_shuffle4(a, b, I) };
+        assert!(c == i64x4(2, 4, 6, 8));
+    }
```

Comment:
What is the expected behavior on passing a negative index?

Reply:
For RMC in principle? CBMC treats these as array indexing, and it would find an array lower bound violation.

For Rust? It requires the indexes are constants, and they're typed `u32`, so it emits an error about the value being out of bounds.

Reply:
Makes sense! Might be useful to add a test with `// rmc-verify-fail` in that case?

Reply:
Well, the "For Rust?" one still applies: I can't get past rustc to show the CBMC bound failure, since it emits compiler errors on attempts to use negative values.

Reply:
Ah sorry I missed that, thanks! Somewhat remarkable, Rust also catches too-high bounds as compile time (I'm used to C-based shuffle intrinsics which tend to dramatically break at runtime):

```rust
        let a = i64x4(1, 2, 3, 4);
        let b = i64x4(5, 6, 7, 8);
        const I: [u32; 4] = [8, 9, 10, 11]; // too big
        let c: i64x4 = unsafe { simd_shuffle4(a, b, I) };
        assert!(c == i64x4(2, 4, 6, 8));
```

```
error[E0511]: invalid monomorphization of `simd_shuffle4` intrinsic: shuffle index #0 is out of bounds (limit 8)
  --> src/main.rs:63:33
   |
63 |         let c: i64x4 = unsafe { simd_shuffle4(a, b, I) };
```